python/lsst/daf/butler/butler.py

```diff
@@ -49,17 +55,95 @@ class Butler:
     ----------
     config : `Config`
         Configuration.
+    collection : `str` or `None`
+        Collection to use for all input lookups, overriding config['collection']
+        if provided.
+    run : `str`, `Run`, or `None`
+        Collection associated with the `Run` to use for outputs, overriding
+        config['run'].  If a `Run` associated with the given Collection does
+        not exist, it will be created.  If "collection" is None, this
+        collection will be used for input lookups as well; if not, it must have
+        the same value as "run".
+
+    Raises
+    ------
+    ValueError
+        Raised if neither 'collection' nor 'run' are provided by argument or
+        config, or if both are provided and are inconsistent.
     """
 
-    def __init__(self, config=None):
+    @staticmethod
+    def makeRepo(root, config=None, expand=False):
```

Comment:
I am wondering whether we should add `searchPaths` here to match those I added to the `ConfigSubset` constructor (and which I didn't add to `ButlerConfig` but should). At the very least it helps testing.

Reply:
I'm actually more worried about flexibility in search paths making it easy to use the wrong search path, and I'm content to leave testing of the search path functionality to lower-level tests for now.